File: sdk/java/src/main/java/vn/hitechcloud/sdk/resources/Ssl.java
package vn.hitechcloud.sdk.resources;

import java.util.Map;
import vn.hitechcloud.sdk.HttpClient;

public class Ssl extends BaseResource {

    public Ssl(HttpClient http) {
        super(http);
    }

    /** List certificates */
    public Map<String, Object> certificates() throws Exception {
        return http.get("/api/certificate");
    }

    /** Get certificate */
    public Map<String, Object> certificate(int id) throws Exception {
        return http.get(String.format("/api/certificate/%id", id));
    }

    /** Get certificate CRT */
    public Map<String, Object> certificateCrt(int id) throws Exception {
        return http.get(String.format("/api/certificate/%id/crt", id));
    }

    /** List SSL orders */
    public Map<String, Object> orders() throws Exception {
        return http.get("/api/certificate/order");
    }

    /** Create SSL order */
    public Map<String, Object> createOrder(Map<String, Object> data) throws Exception {
        return http.post("/api/certificate/order", data);
    }

    /** Get order software */
    public Map<String, Object> orderSoftware(int productId) throws Exception {
        return http.get(String.format("/api/certificate/order/%productId/software", productId));
    }

}

File: sdk/nodejs/src/resources/ssl.js
"use strict";

const { BaseResource } = require("./base");

class SSLResource extends BaseResource {
  async list(params = {}) {
    return this._http.get("/api/certificate");
  }

  async get(certId) {
    return this._http.get("/api/certificate/{id}");
  }

  async getCrt(certId) {
    return this._http.get("/api/certificate/{id}/crt");
  }

  async listOrders(params = {}) {
    return this._http.get("/api/certificate/order");
  }

  async createOrder(data) {
    return this._http.post("/api/certificate/order", data);
  }

  async getOrderSoftware(productId) {
    return this._http.get("/api/certificate/order/{product_id}/software");
  }

}

module.exports = { SSLResource };
